src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2660,6 +2691,20 @@ private void EmitMethodTokenExpression(BoundMethodToken node)
             EmitSymbolToken(node.Method, node.Syntax, null, encodeAsRawToken: true);
         }
 
+        private void EmitMVIDLoad(BoundModuleVersionId node)
+        {
+            Debug.Assert(node.Type.TypeKind == TypeKind.Struct);
+            _builder.EmitOpCode(ILOpCode.Ldsfld);
+            _builder.EmitToken(_module.PrivateImplClass.GetMVID((NamedTypeSymbol)node.Type), node.Syntax, _diagnostics);
```

Comment:
The cast to NamedTypeSymbol is a suspect. We probably need to go from symbol to CCI.IReference, this should be  done through translation. See other call-sites of _builder.EmitToken.